src/snippet_selection.c

```
@@ -121,26 +121,25 @@ void biggest_snippet(scan_data_t *scan)
 		if (j < 0)
 			continue;
 		
-		if (scan->matchmap[j].hits >= min_match_hits) /* Only consider file with more than min_match_hits */
+		if (scan->matchmap[j].hits >= scan->snippet_min_hits) /* Only consider file with more than min_match_hits */
 		{
 			match_data_t *match_new = calloc(1, sizeof(match_data_t)); /* Create a match object */
 			memcpy(match_new->file_md5, scan->matchmap[j].md5, oss_file.key_ln);
```

Cppcheck (2.19.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)